netbox-prometheus-sd.py

```diff
@@ -97,5 +189,14 @@ def main(args):
     parser.add_argument('token', help='Authentication Token')
     parser.add_argument('output', help='Output file')
 
+    parser.add_argument('-d', '--discovery', default='device',
+                        help='Discovery type, default: device', choices=['device', 'circuit'])
     args = parser.parse_args()
-    main(args)
+
+    discovery = Discovery(args)
+
+    discovery.run()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
I prefer to keep the argument parsing in the "if __name__ == '__main__'" block (as a general rule) because the argument parsing is never relevant when the module is imported. On the other hand, calling the main function by providing the already-parsed args can be useful in some cases (eg: create another executable that takes the arguments from environment variables). I know this is less true since you moved all the logic into a class but if you want to change that, please motivate it and separate it into an another commit.